### ISSUE TITLE
membw: compilation error when avx512 is not supported by gcc

### ISSUE THREAD

**Issue by xiaochenshen** [intel/intel-cmt-cat#156]
**Issue: membw compilation error when avx512 is not supported by gcc:**

This issue breaks the building of intel-cmt-cat.
The compilation error:
cc: error: unrecognized command line option ‘-mavx512f’

**1. Test platform: Skylake server**

CPU supports avx512:
```
$ lscpu | grep avx512f

Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch epb cat_l3 cdp_l3 invpcid_single intel_pt ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke avx512_vnni md_clear spec_ctrl intel_stibp flush_l1d arch_capabilities
```

**2. GCC version: 4.8.5**

Avx512 is not supported by gcc 4.8.5:
```
$ gcc -march=native -dM -E - < /dev/null | grep "AVX512F"
```

**3. Building log:**
```
$ make

make -C tools/membw
make[1]: Entering directory `/home/xiaochen/intel-cmt-cat/tools/membw'
cc -MM -MP -MF membw.d -W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wold-style-definition -Wpointer-arith -Wcast-qual -Wundef -Wwrite-strings -Wformat -Wformat-security -fstack-protector -fPIE -Wunreachable-code -Wsign-compare -Wno-endif-labels -Winline -O3 -g -D_FORTIFY_SOURCE=2 -mavx512f -fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv -fno-expensive-optimizations membw.c
cc: error: unrecognized command line option ‘-mavx512f’
cc -W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wold-style-definition -Wpointer-arith -Wcast-qual -Wundef -Wwrite-strings -Wformat -Wformat-security -fstack-protector -fPIE -Wunreachable-code -Wsign-compare -Wno-endif-labels -Winline -O3 -g -D_FORTIFY_SOURCE=2 -mavx512f -fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv -fno-expensive-optimizations   -c -o membw.o membw.c

cc: error: unrecognized command line option ‘-mavx512f’
make[1]: *** [membw.o] Error 1
make[1]: Leaving directory `/home/xiaochen/intel-cmt-cat/tools/membw'
make: *** [all] Error 2
```

**Comment by kmabbasi** [comment]
Thanks Xiaochen for submitting PR.

Just a heads up that we will review it after release 4.0 which is due in WW26.

Thanks,
Khawar

**Comment by xiaochenshen** [comment]
Closed for PR #157 has been merged.
